RNTester/Podfile

```diff
@@ -17,15 +17,69 @@ def pods()
   # Additional Pods which aren't included in the default Podfile
   pod 'React-ART', :path => '../Libraries/ART'
   pod 'React-RCTPushNotification', :path => '../Libraries/PushNotificationIOS'
+  pod 'Yoga',:path => '../ReactCommon/yoga', :modular_headers => true
 
   # Additional Pods which are classed as unstable
   #
   # To use fabric: add `fabric_enabled` option to the use_react_native method above, like below
   # use_react_native!(path: "..", fabric_enabled: true)
 end
 
+def flipper_pods()
+  flipperkit_version = '0.30.1'
+  pod 'FlipperKit', '~>' + flipperkit_version, :configuration => 'Debug'
+  pod 'FlipperKit/FlipperKitLayoutPlugin', '~>' + flipperkit_version, :configuration => 'Debug'
+  pod 'FlipperKit/SKIOSNetworkPlugin', '~>' + flipperkit_version, :configuration => 'Debug'
+  pod 'FlipperKit/FlipperKitUserDefaultsPlugin', '~>' + flipperkit_version, :configuration => 'Debug'
+  pod 'FlipperKit/FlipperKitReactPlugin', '~>' + flipperkit_version, :configuration => 'Debug'
+
+  if ENV['USE_FRAMEWORKS'] == '1'
+    $static_framework = ['FlipperKit', 'Flipper', 'Flipper-Folly',
+        'CocoaAsyncSocket', 'ComponentKit', 'Flipper-DoubleConversion',
+        'Flipper-Glog', 'Flipper-PeerTalk', 'Flipper-RSocket',
+        'CocoaLibEvent', 'OpenSSL-Universal', 'boost-for-react-native']
+
+    pre_install do |installer|
+      Pod::Installer::Xcode::TargetValidator.send(:define_method, :verify_no_static_framework_transitive_dependencies) {}
+      installer.pod_targets.each do |pod|
+          if $static_framework.include?(pod.name)
+            def pod.build_type;
+              Pod::Target::BuildType.static_library
+            end
+          end
+        end
+    end
+  end
```

Comment:
@priteshrnandgaonkar Could you link me to a write-up, if one exists, on the need for these deps to be linked statically?

Reply:
Go through the iOS setup mentioned [here](https://fbflipper.com/docs/getting-started.html)

Reply:
Go through the iOS setup mentioned [here](https://fbflipper.com/docs/getting-started.html)

Reply:
Thanks. I’ve seen those before, but they do not describe _why_ these dependencies need to be linked statically and why a monkey-patch of CocoaPods is required. I would like to understand those technical reasons, as I would like to see if there’s a way to avoid the need to monkey-patch.

Reply:
The reason being, Flipper relies on CocoaLibEvent which has bunch of statically linked binaries libevent.a, libevevnt_core.a etc. due to which the SubPods relying on it has to be built as a Static Framework, which is Flipper-Folly. So ideally speaking building just CocoaLibEvent, Flipper-Folly as a static framework should work, but for some reason the dynamic framework failed to link Flipper-Folly, which I was not able to solve. Thus I built all the Flipper and Flipper related deps as a Static Framework.

Even I feel that we shouldn't have a monkey-patch, if you can solve that problem, then it would be great.

Reply:
Ok, when I’m back from vacation I’ll take a look-see 👍